flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/processor/ResetTransformationProcessorTest.java

```diff
@@ -0,0 +1,157 @@
+                        + " 'connector' = 'values',\n"
+                        + " 'bounded' = 'true'\n"
+                        + ")");
+
+        tEnv.executeSql(
+                "CREATE TABLE Source2 (\n"
+                        + "  a BIGINT,\n"
+                        + "  b BIGINT,\n"
+                        + "  c VARCHAR,\n"
+                        + "  d BIGINT\n"
+                        + ") WITH (\n"
+                        + " 'connector' = 'values',\n"
+                        + " 'bounded' = 'true'\n"
+                        + ")");
+    }
+
+    @Test
+    public void testResetTransformation() {
+        String query = "SELECT * FROM Source1 WHERE a > 100";
+        ExecNodeGraph execNodeGraph = createExecNodeGraph(query);
+        execNodeGraph.getRootNodes().forEach(node -> node.translateToPlan(util.getPlanner()));
+        assertAllTransformationsIsNotNull(execNodeGraph);
+
+        ResetTransformationProcessor resetTransformationProcessor =
+                new ResetTransformationProcessor();
+        resetTransformationProcessor.process(
+                execNodeGraph, new ProcessorContext(util.getPlanner()));
+        assertAllTransformationsIsNull(execNodeGraph);
+    }
+
+    @Test
+    public void testResetTransformationWithExecMultipleInputInExecGraph() {
+        String query =
+                "SELECT Source1.a FROM Source1, Source2 "
+                        + "WHERE Source1.a = Source2.a AND Source2.a = (SELECT Source2.a FROM Source2 WHERE b > 100)";
+        ExecNodeGraph execNodeGraph = createExecNodeGraph(query);
+
+        MultipleInputNodeCreationProcessor multipleInputNodeCreationProcessor =
+                new MultipleInputNodeCreationProcessor(false);
+        multipleInputNodeCreationProcessor.process(
+                execNodeGraph, new ProcessorContext(util.getPlanner()));
+        execNodeGraph.getRootNodes().forEach(node -> node.translateToPlan(util.getPlanner()));
+        assertAllTransformationsIsNotNull(execNodeGraph);
+
+        // If execNodeGraph include batchExecMultipleInput node. The ResetTransformationProcessor
+        // need to both set the transformation in each execNode and the transformation in
+        // batchExecMultipleInput.rootNode to null.
+        ResetTransformationProcessor resetTransformationProcessor =
+                new ResetTransformationProcessor();
+        resetTransformationProcessor.process(
+                execNodeGraph, new ProcessorContext(util.getPlanner()));
+        assertAllTransformationsIsNull(execNodeGraph);
+    }
+
+    private ExecNodeGraph createExecNodeGraph(String query) {
+        Table table = tEnv.sqlQuery(query);
+        RelNode relNode = TableTestUtil.toRelNode(table);
+        FlinkPhysicalRel optimizedRel = (FlinkPhysicalRel) util.getPlanner().optimize(relNode);
+        ExecNodeGraphGenerator generator = new ExecNodeGraphGenerator();
+        return generator.generate(Collections.singletonList(optimizedRel), false);
+    }
```

Comment:
extract a method into `TableTestUtilBase` and refactor  MultipleInputNodeCreationProcessorTest#assertChainableSource

Reply:
> extract a method into `TableTestUtilBase` and refactor MultipleInputNodeCreationProcessorTest#assertChainableSource

Done!